cognite_toolkit/_cdf_tk/prototypes/setup_asset_loader.py

```diff
@@ -0,0 +1,208 @@
+import json
+import re
+import sys
+from collections.abc import Hashable, Iterable
+from functools import lru_cache
+from pathlib import Path
+from typing import Any, Literal
+
+from cognite.client.data_classes import FileMetadataWriteList, TimeSeriesWriteList
+
+import cognite_toolkit._cdf_tk.loaders as loaders
+from cognite_toolkit._cdf_tk._parameters import ANY_INT, ParameterSpec, ParameterSpecSet
+from cognite_toolkit._cdf_tk.exceptions import ToolkitFileNotFoundError
+from cognite_toolkit._cdf_tk.loaders import (
+    LOADER_BY_FOLDER_NAME,
+    LOADER_LIST,
+    RESOURCE_LOADER_LIST,
+    FileMetadataLoader,
+    ResourceLoader,
+    TimeSeriesLoader,
+)
+from cognite_toolkit._cdf_tk.prototypes.resource_loaders import AssetLoader
+from cognite_toolkit._cdf_tk.utils import CDFToolConfig, load_yaml_inject_variables
+
+if sys.version_info >= (3, 10):
+    from typing import TypeAlias
+else:
+    from typing_extensions import TypeAlias
+
+ResourceTypes: TypeAlias = Literal[
+    "assets",
+    "auth",
+    "data_models",
+    "data_sets",
+    "labels",
+    "transformations",
+    "files",
+    "timeseries",
+    "timeseries_datapoints",
+    "extraction_pipelines",
+    "functions",
+    "raw",
+    "workflows",
+]
+
+
+def setup_asset_loader() -> None:
```

Comment:
This is a trick to be able to introduce the `AssetLoader` behind a feature flag. If the flag is turned on, this code modifies the TimeSeriesLoader and FileMetadataLoaders to account for assets. However, if the flag is turned off, these loaders stay as they are, and thus no risk of introducing new bugs in them.